File: clients/kmp/shared/src/commonMain/kotlin/org/wyrdsekai/app/engine/persistence/SoulManifestStore.kt
package org.wyrdsekai.app.engine.persistence

import org.wyrdsekai.app.engine.soul.ClientSoulManifest

/**
 * Persistence for soul manifests on the client.
 * Platform-specific implementations can use SQLite, files, etc.
 * The in-memory implementation is provided for testing.
 */
interface SoulManifestStore {
    suspend fun save(manifest: ClientSoulManifest)
    suspend fun load(did: String): ClientSoulManifest?
    suspend fun delete(did: String)
    suspend fun listDids(): List<String>
}

File: clients/rn/src/engine/persistence/SoulManifestStore.ts
/**
 * Persistence for soul manifests.
 * Mirrors the VitalityStore pattern — interface + in-memory + AsyncStorage impls.
 */

import type { ClientSoulManifest } from '../soul/SoulManifest';

export interface SoulManifestStore {
  save(manifest: ClientSoulManifest): Promise<void>;
  load(did: string): Promise<ClientSoulManifest | null>;
  delete(did: string): Promise<void>;
  listDids(): Promise<string[]>;
}
